framework/tst/dslabs/framework/testing/junit/BaseJUnitTest.java

```diff
@@ -308,7 +308,7 @@ private void assertEndConditionValid() {
                 }
             }, "VizClient");
             thread.setDaemon(false);
-            thread.start();
+            Runtime.getRuntime().addShutdownHook(thread);
```

Comment:
Is there a case when more work will be done after we `throw new VizClientStarted()`? I'm wondering if this should instead be something like
```
vizThread.start();
Runtime.getRuntime().addShutdownHook(() -> vizThread.join());
```

Reply:
Moot not, but that wouldn't do anything. `vizThread` would get killed as soon as `System.exit` was called, and the shutdown hook would finish instantly.